Yosemite/Yosemite/Internal/NSOrderedSet+Array.swift

```diff
@@ -8,3 +8,20 @@ extension NSOrderedSet {
         return array
     }
 }
+
+/// Explicitely checks the type for each generic element that will be converted from NSOrderedSet to a Swift array.
+/// Crash and log the error in debug builds if there's a miss-match
+extension NSOrderedSet {
+    func toTypeCheckedArray<T>() -> [T] {
+        var typeCheckedArray = [T]()
+
+        for element in self {
+            guard let typeCheckedElement = element as? T else {
+                assert(false, "Failed conversion. \(element) is not of type \(T.self)")
```

Comment:
For crashing in debug builds on a developer error, I think `assertionFailure` is used more widely in the code base